src/hooks/Guilds/contracts/useDecodedCallRichContractData.ts

```diff
@@ -0,0 +1,32 @@
+import { DecodedCall } from 'old-components/Guilds/ActionsBuilder/types';
+import { useMemo } from 'react';
+import { RichContractFunction } from './useRichContractData';
+
+const useDecodedCallRichContractData = (decodedCall: DecodedCall) => {

```

Comment:
What do you think of `useRichContractData` here and then maybe `useRichContractConfig` for the hook that fetches from ipfs, the names are a little complex like this I think